### ISSUE TITLE
[ENH] bidsSpatialPrepro re-run in MNI space with normalisation step only

### ISSUE THREAD

**Issue by CerenB** [cpp-lln-lab/bidspm#262]
I think / need this code to do re-running MNI space in bidsSpatialPrepro ONLY the normalisation step IF other steps were already performed in native space.

- to create a workflow that does just the normalization step of a dataset that has already been preprocessed in native space

- make sure that bidsSpatialPrepro can do both MNI and individual preprocessing if requested.

for example in fMRIprep you can ask the output to be in 1, 2, 3 or 4 different spaces and you only run the pipeline once.

If I can have some guidance, I am happy to work on it.

**Comment by Remi-Gau** [comment]
Well, I think this should be broken up in 2 PRs.

1. one for running the normalization of data already processed in native space
1. one for allowing to run preprocessing in native and MNI space

For 2 some (most?) of the work will need to happen in there: 

https://github.com/cpp-lln-lab/CPP_SPM/blob/448182a94ae73c794c8896fbb2fbbe6ea67a89f3/src/batches/setBatchNormalizationSpatialPrepro.m#L3

This is the function that sets up the batch for the normalization in the spatial preprocessing workflow: ss you might see it has been written as part of a workflow with references to some other batches in the workflow as dependencies (using the `cfg_dep` function).

So basically we have to make this function more general and be able to work "on its own" . Not sure it is possible or simple so maybe an easier way to start is to write a separate function to do that and see if we can refactor things later.

If we write a separate function then I suspect it will end up looking a lot like the `setBatchRealign` function because we will have to loop through sessions and runs to list all the images to normalize.

 https://github.com/cpp-lln-lab/CPP_SPM/blob/448182a94ae73c794c8896fbb2fbbe6ea67a89f3/src/batches/setBatchRealign.m#L3

The way I would do it is to generate a batch manually by clicking through the GUI and then try to script it by adapting from `setBatchRealign` to get you roughly there.

Does that make sense?

**Comment by Remi-Gau** [comment]
>     2\. one for allowing to run preprocessing in native and MNI space

For this I think we can do it later and it should be simpler to implement.

**Comment by CerenB** [comment]
slowly getting there but before opening a separated issue in a related matter, do we want `wmean-` functional image? Since when we run `SpatialPrepro`, it is creating `wmean` image but then, we cannot fetch that image by using `getMeanFuncFilename.m` because it looks at `getPrefix` function below part:

```
case 'mean'
      prefix = getPrefix('realign', opt);

      if opt.realign.useUnwarp
        prefix = [spm_get_defaults('unwarp.write.prefix') prefix];
      end

      prefix = ['mean' prefix];
....
```

Here maybe opt.space option can be added? what do you think? 


`getPrefix` is [here](https://github.com/cpp-lln-lab/CPP_SPM/blob/master/src/getPrefix.m). 
`getMeanFuncFilename` is [here](https://github.com/cpp-lln-lab/CPP_SPM/blob/master/src/getMeanFuncFilename.m).

**Comment by Remi-Gau** [comment]
> * to create a workflow that does just the normalization step of a dataset that has already been preprocessed in native space
> 
> * make sure that bidsSpatialPrepro can do both MNI and individual preprocessing if requested

That second point should be addressed by #395: default behavior will now be to run spatial preprocessing in both native and MNI space:

```
opt.space = {'individual', 'MNI'}
```

**Comment by Remi-Gau** [comment]
don't think I will have the time to implement this
